gefserver/pier/internal/dckr/dckr.go

```diff
@@ -280,27 +280,48 @@ func (c Client) GetSwarmContainerInfo(serviceID string) (string, swarm.TaskState
 }
 
 // IsSwarmContainerNotRunning checks if a task container is not running
-func (c Client) IsSwarmContainerNotRunning(serviceID string) (bool, swarm.TaskState, error) {
-	var containerError error
-	swarmTasks, containerError := c.c.ListTasks(docker.ListTasksOptions{})
-	if containerError != nil {
-		return false, swarm.TaskState(""), containerError
+func (c Client) IsSwarmContainerNotRunning(containerID string) (bool, swarm.TaskState, error) {
+	swarmTasks, err := c.c.ListTasks(docker.ListTasksOptions{})
+	if err != nil {
+		return false, swarm.TaskState(""), err
 	}
 
 	for _, task := range swarmTasks {
-		if task.ServiceID == serviceID {
+		if task.Status.ContainerStatus.ContainerID == containerID {
```

Comment:
make a function to get the correct container (findSwarmContainer), then call it here and in the next function